### PR TITLE
GSoC blog post 2: ruma-events breaking changes

### PR DESCRIPTION
**Pull request by DevinR528**
I'm not sure about the code example as it is a bit out of place?

### CODE REVIEW

File: content/news/2020-06-??-gsoc-update-2020.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
++++
+title = "Breaking ruma-events"
+date = 2020-06-01
+extra.author = "Devin Ragotzy"
++++
+
+What a busy week! The ruma homeserver repo was replaced by a monorepo to house all the ruma crates. The 0.22 release of ruma-events will be merged any day now. With Jonas's help and patients, we have made excellent progress! The event kinds are all defined! Using a custom derive to implement the necessary traits for serialization/deserialization they are ready to be sent and received! All but redaction and presence events have a generic content field with a trait bound limiting the field to only valid content types.
```
Comment (jplatte) [inline content/news/2020-06-??-gsoc-update-2020.md]: s/patients/patience :laughing:

Also, even among an audience of people who know Matrix, I'm not sure "event kinds" is going to mean anything to anybody. 'custom derive' should probably also be 'custom derive macro for clarity.

File: content/news/2020-06-??-gsoc-update-2020.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    /// Any message event.
+    Message(MessageEvent<AnyMessageEventContent>),
+    /// `"m.room.redaction"`, on of the non generic event structs.
+    Redaction(RedactionEvent),
+    /// Any state event.
+    State(StateEvent<AnyStateEventContent>),
+}
+```
+
+Jonas started moving ruma-events into ruma/ruma today (10-06-2020) 🎉! Lets hope all goes smoothly.
```
Comment (jplatte) [inline content/news/2020-06-??-gsoc-update-2020.md]: Already done :)

File: content/news/2020-06-??-gsoc-update-2020.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
++++
+title = "Breaking ruma-events"
+date = 2020-06-01
+extra.author = "Devin Ragotzy"
++++
+
+What a busy week! The ruma homeserver repo was replaced by a monorepo to house all the ruma crates. The 0.22 release of ruma-events will be merged any day now. With Jonas's help and patients, we have made excellent progress! The event kinds are all defined! Using a custom derive to implement the necessary traits for serialization/deserialization they are ready to be sent and received! All but redaction and presence events have a generic content field with a trait bound limiting the field to only valid content types.
+```rust
```
Comment (jplatte) [inline content/news/2020-06-??-gsoc-update-2020.md]: I'm fine with code blocks in general, but this one is a bit long.. Not sure what to do with it.

**Review by jplatte — COMMENTED** [review]
Very rough first review.

File: content/news/2020-06-??-gsoc-update-2020.md (hunk 1 above)
Comment (DevinR528) [reply content/news/2020-06-??-gsoc-update-2020.md]: > The event kinds (state, message, ephemeral, etc.)...

How's this ^.

File: content/news/2020-06-??-gsoc-update-2020.md (hunk 3 above)
Comment (DevinR528) [reply content/news/2020-06-??-gsoc-update-2020.md]: If this is still to big maybe we could do a dropdown?

**Comment by jplatte** [comment]
Since this is meant to document your GSoC project, could you focus on what you did rather than the overall progress? We can have the latter in the general TWiM section / a later blog post though.

File: content/news/2020-06-11-gsoc-update-2020.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
++++
+title = "Breaking ruma-events"
+date = 2020-06-01
+extra.author = "Devin Ragotzy"
++++
+
+This week in ruma-events' Google Summer of Code project, I was able to finish the macros needed to generate the event content enums and trait implementations required for events. We started by defining the generic event structs (state, message, etc.) and manually writing the `Serialize` and `Deserialize` implementations. Over the next few days, this was moved into a custom derive macro called `Event`. The derive now implements all necessary traits with appropriate bounds, so a `StateEvent<C>` can not contain any ephemeral event content and so forth. We have removed the `raw` mod and related `FromRaw` and `TryFromRaw` traits, moving the validation into the deserialization and constructor for the few types that needed it. On the event content side of things, a function like procedural macro was used to allow declaring the enum using Matrix event type identifiers.
```
Comment (jplatte) [inline content/news/2020-06-11-gsoc-update-2020.md]: I'm guessing the wording 'We' is left over from the previous iteration?

```suggestion
This week in ruma-events' Google Summer of Code project, I was able to finish the macros needed to generate the event content enums and trait implementations required for events. I started by defining the generic event structs (state, message, etc.) and manually writing the `Serialize` and `Deserialize` implementations. Over the next few days, this was moved into a custom derive macro called `Event`. The derive now implements all necessary traits with appropriate bounds, so a `StateEvent<C>` can not contain any ephemeral event content and so forth. I have removed the `raw` mod and related `FromRaw` and `TryFromRaw` traits, moving the validation into the deserialization and constructor for the few types that needed it. On the event content side of things, a function like procedural macro was used to allow declaring the enum using Matrix event type identifiers.
```

**Review by jplatte — COMMENTED** [review]
:+1:

One more suggestion for you: Link to your PRs. I'm personally a fan of just adding links to parts of the sentence, as in

> I started by \[adding a foo to `Bar`\]\(link\), after which `Baz` could be refactored to …

But

> I started adding a foo to `Bar` (\[repo#123\]\(link\))

or footnotes, or whatever, would be fine too.

File: content/news/2020-06-??-gsoc-update-2020.md (hunk 1 above)
Comment (jplatte) [reply content/news/2020-06-??-gsoc-update-2020.md]: Sounds good.

**Review by jplatte — APPROVED** [review]

**Pull request merged**